artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireMessageConverter.java

```diff
@@ -590,9 +593,14 @@ private static ActiveMQMessage toAMQMessage(MessageReference reference,
       }
       amqMsg.setCommandId(commandId);
 
-      final SimpleString corrId = getObjectProperty(coreMessage, SimpleString.class, OpenWireConstants.JMS_CORRELATION_ID_PROPERTY);
-      if (corrId != null) {
-         amqMsg.setCorrelationId(corrId.toString());
+      final Object correlationID = coreMessage.getCorrelationID();
+      if (correlationID != null) {
+         if (correlationID instanceof String || correlationID instanceof SimpleString) {
```

Comment:
Can the core message carry something else besides SimpleString, String or byte[] and if so should it be ignored as it is now?  If not should the if be inverted to simply check for byte[] and just toString anything else?